test/unitTest/java/org/thoughtcrime/securesms/compression/CompressionTest.java

```diff
@@ -0,0 +1,117 @@
+        PowerMockito.when(mockCompressor.compressVideo(anyString(), anyString())).thenReturn("test/unitTest/java/org/thoughtcrime/securesms/resources/test-video-compressed.mp4");
+        PowerMockito.whenNew(SiliCompressor.class).withAnyArguments().thenReturn(mockCompressor);
+    }
+
+    @Test
+    public void testCompressImageFile() throws Exception {
+        File testImg = new File("test/unitTest/java/org/thoughtcrime/securesms/resources/test-img.jpg");
+        String testDirectory = testImg.getPath();
+
+        MediaConstraints temp = MediaConstraints.getPushMediaConstraints();
+
+        String compressedPath = temp.compressImage(context, testDirectory);
+
+        Long uncompressedLength = testImg.length();
+        Long compressedLength = new File(compressedPath).length();
+
+        assertTrue(compressedLength < uncompressedLength);
+    }
+
+    @Test
+    public void testCompressVideo() throws Exception {
+        Context mockContext = mock(Context.class);
+        when(mockContext.getCacheDir()).thenReturn(mock(File.class));
+
+        File testVideo = new File("test/unitTest/java/org/thoughtcrime/securesms/resources/test-video.mp4");
+        String testDirectory = testVideo.getPath();
+
+        MediaConstraints temp = MediaConstraints.getPushMediaConstraints();
+
+        String compressedPath = temp.compressVideo(mockContext, testDirectory);
+
+        Long uncompressedLength = testVideo.length();
+        Long compressedLength = new File(compressedPath).length();
+
+        assertTrue(compressedLength < uncompressedLength);
+    }
+
+    @Test
+    public void testSatisfiesCompressionSetting() throws Exception {
+        Attachment a1 = getMockAttachment("image/jpeg");
+        Attachment a2 = getMockAttachment("image/gif");
+        Attachment a3 = getMockAttachment("video/mp4");
+
+        Set<String> testPreferences = new HashSet<String>();
+        testPreferences.add("image");
+        testPreferences.add("video");
+        testPreferences.add("gif");
+
+        MediaConstraints temp = MediaConstraints.getPushMediaConstraints();
+
+        assertTrue(temp.satisfiesCompression(context, a1, testPreferences));
+        assertTrue(temp.satisfiesCompression(context, a2, testPreferences));
+        assertTrue(temp.satisfiesCompression(context, a3, testPreferences));
+    }
+
+    @Test
+    public void testDoesntSatisfyCompressionSetting() throws Exception {
+        Attachment a1 = getMockAttachment("image/jpeg");
+        Attachment a2 = getMockAttachment("image/gif");
+        Attachment a3 = getMockAttachment("video/mp4");
+
+        Set<String> testPreferences = new HashSet<String>();
+
+        MediaConstraints temp = MediaConstraints.getPushMediaConstraints();
+
+        assertTrue(!temp.satisfiesCompression(context, a1, testPreferences));
+        assertTrue(!temp.satisfiesCompression(context, a2, testPreferences));
+        assertTrue(!temp.satisfiesCompression(context, a3, testPreferences));
+    }
```

Comment:
Can you test the case in which the file does not exist ???

Reply:
cant happen in real world tho cuz compression can only get called on a already created file?